tests/integration/helpers.py

```diff
@@ -266,3 +267,171 @@ def get_job_invoked_unit(job: jenkins.jenkinsapi.job.Job, units: typing.List[Uni
+            <description></description>
+            <keepDependencies>false</keepDependencies>
+            <properties/>
+            <definition
+                class="org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition"
+                plugin="workflow-cps@3837.v305192405b_c0">
+                <script>{script}</script>
+                <sandbox>true</sandbox>
+            </definition>
+            <triggers/>
+            <disabled>false</disabled>
+        </flow-definition>
+        """
+    )
+
+
+def kubernetes_test_pipeline_script() -> str:
+    """Generate a test pipeline script using the kubernetes plugin.
+
+    Return:
+        The pipeline script
+    """
+    return textwrap.dedent(
+        """
+        podTemplate(yaml: '''
+            apiVersion: v1
+            kind: Pod
+            metadata:
+            labels:
+                some-label: some-label-value
+            spec:
+            containers:
+            - name: httpd
+              image: httpd
+              command:
+              - sleep
+              args:
+              - 99d
+              tty: true
+        ''') {
+        node(POD_LABEL) {
+            stage('Integration Test') {
+            sh '''#!/bin/bash
+                hostname
+            '''
+            }
+        }
+        }"""
+    )
+
+
+def create_secret_file_credentials(
+    unit_web_client: UnitWebClient, kube_config: str
+) -> typing.Optional[str]:
+    """Use the jenkins client to create a new secretfile credential.
+    plain-credentials plugin is required.
+
+    Args:
+        unit_web_client: Client for Jenkins's remote access API.
+        kube_config: path to the kube_config file.
+
+    Returns:
+        The id of the created credential, or None in case of error.
+    """
+    url = f"{unit_web_client.web}/credentials/store/system/domain/_/createCredentials"
+    credentials_id = f"kube-config-{secrets.token_hex(4)}"
+    payload = {
+        "json": f"""{{
+            "": "4",
+            "credentials": {{
+                "file": "file0",
+                "id": "{credentials_id}",
+                "description": "Created by API",
+                "stapler-class": "org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl",
+                "$class": "org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl",
+            }},
+        }}"""
+    }
+
+    accept_header = (
+        "text/html,"
+        "application/xhtml+xml,"
+        "application/xml;q=0.9,"
+        "image/avif,image/webp,"
+        "image/apng,"
+        "*/*;q=0.8,"
+        "application/signed-exchange;v=b3;q=0.9'"
+    )
+    headers = {
+        "Accept": accept_header,
+    }
+
+    with open(kube_config, "rb") as kube_config_file:
+        files = [("file0", ("config", kube_config_file, "application/octet-stream"))]
+        logger.debug("Creating jenkins credentials, params: %s %s %s", headers, files, payload)
+        res = unit_web_client.client.requester.post_url(
```

Comment:
would a timeout on the request help?

Reply:
indeed, thanks. I'll update it in a follow-up PR and I'll close this one